### PR TITLE
Fixed #29084 -- Skipped some postgres_tests.test_search tests when pg_catalog isn't English.

### PR DESCRIPTION
**Pull request by Polandia94**
When the conf in postgresql.conf is not default_text_search_config = 'pg_catalog.english' some test fail. Now check the pg_catalog before run the tests.

### CODE REVIEW

File: tests/postgres_tests/__init__.py [hunk 1]
```diff
@@ -17,7 +17,12 @@ class PostgreSQLSimpleTestCase(SimpleTestCase):
 # To register type handlers and locate the widget's template.
 @modify_settings(INSTALLED_APPS={"append": "django.contrib.postgres"})
 class PostgreSQLTestCase(TestCase):
-    pass
+    def check_pg_catalog(self):
+        with connection.cursor() as cursor:
+            cursor.execute("show default_text_search_config")
+            default_config = cursor.fetchone()[0]
+            if default_config != "pg_catalog.english":
+                self.skipTest("Default config is not 'english'")
```
Comment (felixxm) [inline tests/postgres_tests/__init__.py:25]: This is required only for search tests so I'd move it to the `test_search.py`.

File: tests/postgres_tests/__init__.py [hunk 2]
```diff
@@ -17,7 +17,12 @@ class PostgreSQLSimpleTestCase(SimpleTestCase):
 # To register type handlers and locate the widget's template.
 @modify_settings(INSTALLED_APPS={"append": "django.contrib.postgres"})
 class PostgreSQLTestCase(TestCase):
-    pass
+    def check_pg_catalog(self):
+        with connection.cursor() as cursor:
+            cursor.execute("show default_text_search_config")
+            default_config = cursor.fetchone()[0]
```
Comment (felixxm) [inline tests/postgres_tests/__init__.py:23]: Can we change it to the cached property? e.g. `default_text_search_config`.
```python
@cached_property
def default_text_search_config(self):
    with connection.cursor() as cursor:
        cursor.execute("show default_text_search_config")
        row = cursor.fetchone()
        return row[0] if row else None

def has_english_default_text_search_config(self):
    if self.default_text_search_config != "pg_catalog.english":
        self.skipTest("Default text search config is not 'english'")
```

**Review by felixxm — COMMENTED** [review]
@Polandia94 Thanks :+1: I left comments.

File: tests/postgres_tests/test_search.py [hunk 3]
```diff
@@ -358,6 +360,7 @@ def test_vector_add_multi(self):
         )
 
     def test_vector_combined_mismatch(self):
+        self.check_pg_catalog()
```
Comment (felixxm) [inline tests/postgres_tests/test_search.py:363]: This test doesn't fail with `pg_catalog.simple`, please revert.

File: tests/postgres_tests/test_search.py [hunk 4]
```diff
@@ -438,6 +442,7 @@ def test_combined_configs(self):
         self.assertSequenceEqual(searched, [self.verse2])
 
     def test_combine_raw_phrase(self):
+        self.check_pg_catalog()
```
Comment (felixxm) [inline tests/postgres_tests/test_search.py:445]: This test doesn't fail with `pg_catalog.simple`, please revert.

**Review by felixxm — COMMENTED** [review]
As far as I'm aware, `postgres_tests.test_search.SearchHeadlineTests.test_headline_untyped_args` should also be skipped:
```
FAIL: test_headline_untyped_args (postgres_tests.test_search.SearchHeadlineTests)
...
AssertionError: 'Bravely bold Sir Robin, rode forth from C[31 chars]e, o' != 'Robin. He was not at all afraid to be <b>[58 chars]obin'
- Bravely bold Sir Robin, rode forth from Camelot. He was not afraid to die, o
+ Robin. He was not at all afraid to be <b>killed</b> in nasty ways. Brave, brave, brave, brave Sir Robin
```
I have 5 failures with non-English search configuration (e.g. `pg_catalog.french`):
- `test_headline`
- `test_headline_fragments_words_options`
- `test_headline_highlight_all_option`
- `test_headline_untyped_args`
- `test_combine_raw_phrase`

and a different 5 failures with `pg_catalog.simple`:
- `test_headline_highlight_all_option`
- `test_headline_untyped_args`
- `test_non_exact_match`
- `test_search_two_terms`
- `test_combine_different_vector_configs`

**Comment by felixxm** [comment]
I reached 9 failures for `pg_catalog.finnish` :smile:, so we should probably skip the following tests:
- `test_combine_raw_phrase`
- `test_combine_different_vector_configs`
- `test_headline`
- `test_headline_fragments_words_options`
- `test_headline_highlight_all_option`
- `test_headline_short_word_option`
- `test_headline_untyped_args`
- `test_non_exact_match`
- `test_search_two_terms`

**Comment by Polandia94** [comment]
I corrected all the comments (still have problems with the linters,for some reason can run it locally), except for moving from init to test_search.py
The class PostgreSQLTestCase is defined in __init__ and is inherited from the test cases. So i see three options, and i don't think any of this is good:
Define has_english_default_text_search_config in GrailTestData, it will work, but it shouldn'n be there
MonkeyPatch PostgreSQLTestCase
Create a new class in test_search.py, who inherited from PostgreSQLTestCase and has the new methods. This will require change the inheritance from PostgreSQLTestCase to the new class in various test

**Comment by felixxm** [comment]
@Polandia94 Thanks for updates :+1:

I pushed tiny edits.

**Pull request merged**